src/rgw/rgw_admin.cc

```diff
@@ -176,6 +176,7 @@ void usage()
   cout << "  bucket sync checkpoint           poll a bucket's sync status until it catches up to its remote\n";
   cout << "  bucket sync disable              disable bucket sync\n";
   cout << "  bucket sync enable               enable bucket sync\n";
+  cout << "  bucket sync init                 initialize bucket sync indicated by --state flag\n";
```

Comment:
right, I just wanted to re-use the existing code so I just added the state as an input here to extend. I guess we can name it something general like `bucket sync reset-state --state {state}` so we let admin switch from states as wished for now. what do you think?

> but again, this feature would subvert disaster recovery and break symmetry

Is this something that again you think we should control by checking the source within the same zonegroup or are you referring to some other scenario? as we discussed the purpose of this PR is to subvert from DR and break symmetry but perhaps only when the source zone is in a different zonegroup?

Reply:
that's right, thanks. I changed `bucket sync init` to `bucket sync reset-state` with only limited states `init`, `full-sync`, and `incremental-sync`. so this way we won't let stopped pass by that path. Also, made sure `incremental-sync` is not applied to a normal bucket sync scenario (within the zonegroup) to break the symmetry.

> if we skip the full sync, bucket sync init is just going to advance the incremental sync status markers to the end of the remote's bilogs. this means that it'll only replicate new changes that happen after the command runs. is that really the intent?

I believe we need to. As the current implementation, full sync doesn't run in parallel to incremental, In case the state was set to full and that is taking so long or was a mistake and admin want to switch back to incremental, there should be a way to do so otherwise it might stuck forever.

Reply:
> otherwise it might stuck forever

as far as i know, the only way it would get stuck forever is if data sync loses track of the datalog entries that would trigger that bucket sync. a `bucket sync reset-state` wouldn't necessarily help with that either, because it doesn't write the corresponding datalog entries

instead, we have the `bucket sync run` command for this case. it manually processes the rest of bucket sync, including the transition from full to incremental. unlike the proposed `bucket sync reset-state`, this command preserves multisite's guarantee of eventual consistency of the bucket across zones

Reply:
> unlike the proposed bucket sync reset-state

right, but I didn't mean to replace `bucket sync run` with `bucket sync reset-state` (maybe the naming is misleading). my intent here is to open a door to the state machine of sync state for the admin to change it (to init, full-sync, and incremental) based on whatever factor he thinks is required and then eventually run the sync manually with `bucket sync run` or let the normal sync make it.
So the situations are:
- **init**: this is where we want to clear the state to restore consistency of a bucket. 
- **full-sync**: here is where the feature is enabled but we want to simulate the [Batch Replication](https://docs.aws.amazon.com/AmazonS3/latest/userguide/s3-batch-replication-batch.html). For this maybe we should only consider the current state is incremental before the transition.
- **incremental**: this is where the bucket is in full sync but this is taking so long or was a mistake (stuck was bad wording, I agree :) ) and the admin wants to revert it back to incremental (with respect to "normal bucket replication within zonegroup").

Can you share your thoughts on them? should we introduce each as a different command or you think we should not support some of them?